src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.NuGetPackages.cs

```diff
@@ -63,7 +75,7 @@ private bool ContainsDependency(TargetFrameworkMoniker tfm, Func<PackageDependen
 
         private IEnumerable<LockFileTargetLibrary> GetAllDependencies(TargetFrameworkMoniker tfm)
         {
-            if (LockFilePath is null)
+            if (!IsRestored)
             {
                 throw new InvalidOperationException("Project needs to be restored before continuing");
```

Comment:
why throw here instead of `await _restorer.RestorePackagesAsync(Context, this, token).ConfigureAwait(false);`

Reply:
It should always be restored here

Reply:
I like the change to file an issue as an error message. seems like a pattern we should have in other places.